modules/p2/ding_message/ding_message.module

```diff
@@ -278,49 +239,36 @@ function ding_message_check_list_for_updates($uid, $element_id, $mid, $debug = F
     $new_elements_count = 0;
   }
 
-  if ($debug) {
-    drupal_set_message(check_plain('Updating list of type: ' . $type));
-  }
-
-  // Handle the different ding_list types.
-  switch ($type) {
-    case 'ting_object':
-    case 'ding_list_user':
-      // Types not eglible for further processing.
-      break;
-
-    case 'taxonomy_term':
-      module_load_include('inc', 'opensearch', 'open.client');
-      // Get the element value.
-      $value = ding_list_get_list_element_value($element);
-      // Load taxonomy term.
-      $term = taxonomy_term_load($value);
-      // Taxonomy terms updates are a ting search
-      // for materials with the subject = term->name.
-      $result = ting_start_query()
-        ->withFieldFilter(TingSearchCommonFields::SUBJECT, check_plain($term->name))
-        ->withCount(50)
-        ->execute();
+  switch ($element['type']) {
+    case 'follow_author':
+    case 'search_query':
+      module_load_include('inc', 'opensearch', 'opensearch.client');
+      // Build a search query, since the url path doesn't
+      // match the query accepted by ting_do_search.
+      $query = ding_message_build_query($element['value']);
+      // TODO Port to search abstraction layer.
+      $result = $query->withCount(50)->execute();
       if ($result->getNumTotalCollections() > 0) {
         // Get the first element in the search result set.
         $collections = $result->getTingEntityCollections();
         $first_collection = current($collections);
         if (!empty($message)) {
           // If the first search result element
-          // doesn't match the stored last id,
-          // update the message.
+          // doesn't match the stored last id, update the message.
           if ($first_collection->getId() != $last_element_id) {
             // Get the result set ids.
-            $collection_ids = array_map(function(\TingCollection $collection) {
+            $collection_ids = array_map(function (TingCollection $collection) {
               return $collection->getId();
             }, $collections);
             // Get the amount of new elements since last check.
+            // We assume the result set is consistently ranked by date.
```

Comment:
Search sematics have been changed a number of times breaking this fragile assumption (which was cleared with the original p1 team) and leading to incorrect suggestions.
We need to know what is the correct way to perform a query and have the result ordered by publish date in a consistent manner so the order would never change on subsequent queries. As I recall, earlier if we applied specific sorting to the query, then granulatity dropped to publish year which is useless for this purpose. Where do we best ask this question?

Reply:
DBC would probably be the best best. The [OpenSearch documentation](https://opensource.dbc.dk/services/open-search-web-service) says:

> Search results may be sorted ascending or descending by several different parameters, using the sort-parameter. This is done by title, creator or various dates.

It includes `sort=date_descending` in an example. That sorting may suffer from the problem you just described. Sorting by acquisition date should provide a stable order. However that may not possible at the moment.

@JrgenGNielsen: Can you help here?

Reply:
I'd suggest sorting by acquisitionDate_descending

Reply:
According to @cableman sorting by acquisition date is/was not possible?

Reply:
Acquisition date is not optimal because it will return old materials recently acquired.

Another idea is to sort by publish year AND some other stable index like internal primary index or acquisition date.